project-generator/src/main/java/io/syndesis/project/converter/visitor/EndpointStepVisitor.java

```diff
@@ -110,9 +107,9 @@ public EndpointStepVisitor(GeneratorContext generatorContext) {
         //
         final String connectorScheme = hasComponentOptions ? connection.getId().map(id -> camelConnectorPrefix + "-" + id).orElse(camelConnectorPrefix) : camelConnectorPrefix;
 
-        Map<String, String> secrets = connector.filterProperties(
+        Map<String, String> secrets = action.filterProperties(
             properties,
-            connector.isSecret(),
+            action.isSecret(),
```

Comment:
Do we need to check with connection if the property is secret also?
something like `connection.isSecret().or(action.isSecret())`

_Sidenote_
At first method named `is...` got me worried but then I understood that it returns a `Predicate`, wonder if anyone else thinks that we could have method handle here or name those `is...` methods returning `Predicate` differently, something like `test...`?

Reply:
Yeah, will add such check.

Sidenote: 
yeah, there are now two variant of the isSecret, one that returns a predicate and one that can be used as method reference, I kept the predicate one for backward compatibility for the moment.

Reply:
@zregvart should be fixed